### PR TITLE
Add error message in JSON response

### PR DESCRIPTION
**Pull request by jkakavas**
When the SAML authentication is not successful, we return a SAML
Response with a status that indicates a failure. This commit adds
an error message in the REST API response along with the SAML
Response XML string so that the caller of the API can identify
that this is an unsuccessful response without needing to parse the
XML.

### CODE REVIEW

**Comment by jkakavas** [comment]
@elasticmachine update branch

File: x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ public RestResponse buildResponse(SamlInitiateSingleSignOnResponse response, XCo
                         builder.startObject();
                         builder.field("post_url", response.getPostUrl());
                         builder.field("saml_response", response.getSamlResponse());
+                        builder.field("error", response.getError());
```
Comment (tvernum) [inline x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java]: Is the contract that if `error` is `null` then it's a success and if it's non-null then there's a failure?

**Review by tvernum — APPROVED** [review]

File: x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java (hunk 1 above)
Comment (jkakavas) [reply x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java]: yes. I contemplated adding a boolean `isSuccess` field but felt like this extra check wouldn't make much sense on the consumer side but I don't feel strongly about it if you think it makes this clearer/easier to consume.

Comment (tvernum) [reply x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java]: It feels a little weird, but I can't think of anything better, unless we add the saml status as well.

Comment (jkakavas) [reply x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestSamlInitiateSingleSignOnAction.java]: I ended up doing just that after all

**Comment by jkakavas** [comment]
ping @tvernum in case you want to take another look since I also cleaned up the SamlAuthenticationState and added a field to the response as we talked about

**Review by tvernum — APPROVED** [review]

**Comment by jkakavas** [comment]
backport to 7.7 depends on https://github.com/elastic/elasticsearch/pull/54553 being merged

**Pull request merged**